From 49bf9d17f0b3f7ebb4f9f562f10853e3fd067b7a Mon Sep 17 00:00:00 2001
From: Nagendrababu <nagendrababu.bonkuri@connectus.ai>
Date: Wed, 30 Jun 2021 12:46:26 -0400
Subject: [PATCH] WiFi6 Aps LEDs Label name change

This patch fixes leds incorrect state issue and also chanage the label
names of leds

Signed-off-by: Nagendrababu <nagendrababu.bonkuri@connectus.ai>
---
 .../ipq807x/base-files/etc/board.d/01_leds    |  13 +-
 .../111-WiFi6-Aps-LEDs-Label-Change.patch     | 183 ++++++++++++++++++
 2 files changed, 193 insertions(+), 3 deletions(-)
 create mode 100644 target/linux/ipq807x/patches/111-WiFi6-Aps-LEDs-Label-Change.patch

diff --git a/target/linux/ipq807x/base-files/etc/board.d/01_leds b/target/linux/ipq807x/base-files/etc/board.d/01_leds
index 1402a74f3e..bca3ae8daa 100755
--- a/target/linux/ipq807x/base-files/etc/board.d/01_leds
+++ b/target/linux/ipq807x/base-files/etc/board.d/01_leds
@@ -15,10 +15,17 @@ cig,wf194c)
 	ucidef_set_led_netdev "wan" "WAN" "wf194c:green:wan" "br-wan" "tx rx link"
 	ucidef_set_led_netdev "lan" "LAN" "wf194c:green:lan" "br-lan" "tx rx link"
 	;;
-edgecore,eap101 |\
+edgecore,eap101)
+	ucidef_set_led_wlan "wifi5g" "WIFI5G" "eap101:green:wifi5g" "phy0tpt"
+	ucidef_set_led_wlan "wifi2g" "WIFI2G" "eap101:green:wifi2g" "phy1tpt"
+	;;
 edgecore,eap102)
-        ucidef_set_led_wlan "wlan5g" "WLAN5G" "green:wifi5" "phy0tx"
-        ucidef_set_led_wlan "wlan2g" "WLAN2G" "green:wifi2" "phy1tx"
+	ucidef_set_led_wlan "wifi5g" "WIFI5G" "eap102:green:wifi5g" "phy0tpt"
+	ucidef_set_led_wlan "wifi2g" "WIFI2G" "eap102:green:wifi2g" "phy1tpt"
+	;;
+cig,wf188n)
+	ucidef_set_led_wlan "wifi5g" "WIFI5G" "wf188:green:wifi5g" "phy0tpt"
+	ucidef_set_led_wlan "wifi2g" "WIFI2G" "wf188:green:wifi2g" "phy1tpt"
 	;;
 esac
 
diff --git a/target/linux/ipq807x/patches/111-WiFi6-Aps-LEDs-Label-Change.patch b/target/linux/ipq807x/patches/111-WiFi6-Aps-LEDs-Label-Change.patch
new file mode 100644
index 0000000000..06533425a9
--- /dev/null
+++ b/target/linux/ipq807x/patches/111-WiFi6-Aps-LEDs-Label-Change.patch
@@ -0,0 +1,183 @@
+Index: linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce/arch/arm64/boot/dts/qcom/qcom-ipq6018-cig-wf188.dts
+===================================================================
+--- linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce.orig/arch/arm64/boot/dts/qcom/qcom-ipq6018-cig-wf188.dts
++++ linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce/arch/arm64/boot/dts/qcom/qcom-ipq6018-cig-wf188.dts
+@@ -309,28 +309,28 @@
+ 		pinctrl-names = "default";
+ 
+ 		led@25 {
+-			label = "led_5g";
++			label = "wf188:green:wifi5g";
+ 			gpios = <&tlmm 25 GPIO_ACTIVE_HIGH>;
+-			linux,default-trigger = "wf188:green:5g";
++			linux,default-trigger = "wf188:green:wifi5g";
+ 			default-state = "off";
+ 		};
+ 		led@24 {
+-			label = "led_2g";
++			label = "wf188:green:wifi2g";
+ 			gpios = <&tlmm 24 GPIO_ACTIVE_HIGH>;
+-			linux,default-trigger = "wf188:green:2g";
++			linux,default-trigger = "wf188:green:wifi2g";
+ 			default-state = "off";
+ 		};
+ 		led@18 {
+-			label = "led_eth";
++			label = "wf188:green:eth";
+ 			gpios = <&tlmm 18 GPIO_ACTIVE_HIGH>;
+ 			linux,default-trigger = "wf188:green:eth";
+-			default-state = "off";
++			default-state = "on";
+ 		};
+                 led_power: led@16 {
+-                        label = "led_pwr";
++                        label = "wf188:green:power";
+                         gpios = <&tlmm 16 GPIO_ACTIVE_HIGH>;
+                         linux,default-trigger = "wf188:green:power";
+-			default-state = "off";
++			default-state = "on";
+ 		};
+ 	};
+ 
+Index: linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce/arch/arm64/boot/dts/qcom/qcom-ipq6018-cig-wf188n.dts
+===================================================================
+--- linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce.orig/arch/arm64/boot/dts/qcom/qcom-ipq6018-cig-wf188n.dts
++++ linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce/arch/arm64/boot/dts/qcom/qcom-ipq6018-cig-wf188n.dts
+@@ -309,28 +309,28 @@
+ 		pinctrl-names = "default";
+ 
+ 		led@25 {
+-			label = "led_5g";
++			label = "wf188:green:wifi5g";
+ 			gpios = <&tlmm 25 GPIO_ACTIVE_HIGH>;
+-			linux,default-trigger = "wf188:green:5g";
++			linux,default-trigger = "wf188:green:wifi5g";
+ 			default-state = "off";
+ 		};
+ 		led@24 {
+-			label = "led_2g";
++			label = "wf188:green:wifi2g";
+ 			gpios = <&tlmm 24 GPIO_ACTIVE_HIGH>;
+-			linux,default-trigger = "wf188:green:2g";
++			linux,default-trigger = "wf188:green:wifi2g";
+ 			default-state = "off";
+ 		};
+ 		led@18 {
+-			label = "led_eth";
++			label = "wf188:green:eth";
+ 			gpios = <&tlmm 18 GPIO_ACTIVE_HIGH>;
+ 			linux,default-trigger = "wf188:green:eth";
+-			default-state = "off";
++			default-state = "on";
+ 		};
+                 led_power: led@16 {
+-                        label = "led_pwr";
++                        label = "wf188:green:power";
+                         gpios = <&tlmm 16 GPIO_ACTIVE_HIGH>;
+                         linux,default-trigger = "wf188:green:power";
+-			default-state = "off";
++			default-state = "on";
+ 		};
+ 	};
+ 
+Index: linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce/arch/arm64/boot/dts/qcom/qcom-ipq6018-edgecore-eap101.dts
+===================================================================
+--- linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce.orig/arch/arm64/boot/dts/qcom/qcom-ipq6018-edgecore-eap101.dts
++++ linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce/arch/arm64/boot/dts/qcom/qcom-ipq6018-edgecore-eap101.dts
+@@ -337,21 +337,21 @@
+ 		pinctrl-names = "default";
+ 
+ 		led@25 {
+-			label = "green:wifi5";
+-			gpios = <&tlmm 35 GPIO_ACTIVE_LOW>;
+-			linux,default-trigger = "wf188:green:5g";
++			label = "eap101:green:wifi5g";
++			gpios = <&tlmm 35 GPIO_ACTIVE_HIGH>;
++			linux,default-trigger = "eap101:green:wifi5g";
+ 			default-state = "off";
+ 		};
+ 		led@24 {
+-			label = "green:wifi2";
+-			gpios = <&tlmm 37 GPIO_ACTIVE_LOW>;
+-			linux,default-trigger = "wf188:green:2g";
++			label = "eap101:green:wifi2g";
++			gpios = <&tlmm 37 GPIO_ACTIVE_HIGH>;
++			linux,default-trigger = "eap101:green:wifi2g";
+ 			default-state = "off";
+ 		};
+                 led_power: led@16 {
+-                        label = "led_pwr";
++                        label = "eap101:green:power";
+                         gpios = <&tlmm 74 GPIO_ACTIVE_HIGH>;
+-                        linux,default-trigger = "green:power";
++                        linux,default-trigger = "eap101:green:power";
+ 			default-state = "off";
+ 		};
+ 	};
+Index: linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce/arch/arm64/boot/dts/qcom/qcom-ipq807x-eap102.dts
+===================================================================
+--- linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce.orig/arch/arm64/boot/dts/qcom/qcom-ipq807x-eap102.dts
++++ linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce/arch/arm64/boot/dts/qcom/qcom-ipq807x-eap102.dts
+@@ -671,29 +671,29 @@
+ 		pinctrl-names = "default";
+ 
+ 			led_power: led_pwr {
+-				label = "green:power";
++				label = "eap102:green:power";
+ 				gpios = <&tlmm 46 GPIO_ACTIVE_HIGH>;
+ 				default-state = "on";
+-				linux,default-trigger = "led_pwr";
++				linux,default-trigger = "eap102:green:power";
+ 			};
+ 
+ 			led_2g {
+-				label = "green:wifi2";
++				label = "eap102:green:wifi2g";
+ 				gpio = <&tlmm 47 GPIO_ACTIVE_HIGH>;
+ 				default-state = "off";
+ 			};
+ 
+ 			led_5g {
+-				label = "green:wifi5";
++				label = "eap102:green:wifi5g";
+ 				gpio = <&tlmm 48 GPIO_ACTIVE_HIGH>;
+ 				default-state = "off";
+ 			};
+ 
+ 			led_bt {
+ 				gpios = <&tlmm 50 GPIO_ACTIVE_HIGH>;
+-				label = "green:bt";
+-				default-state = "off";
+-				linux,default-trigger = "led_bt";
++				label = "eap102:green:bt";
++				default-state = "on";
++				linux,default-trigger = "eap102:green:bt";
+ 			};
+ 	};
+ 	nss-macsec0 {
+Index: linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce/arch/arm64/boot/dts/qcom/qcom-ipq807x-ex227.dts
+===================================================================
+--- linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce.orig/arch/arm64/boot/dts/qcom/qcom-ipq807x-ex227.dts
++++ linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce/arch/arm64/boot/dts/qcom/qcom-ipq807x-ex227.dts
+@@ -600,7 +600,7 @@
+ 		pinctrl-names = "default";
+ 
+ 		led_power {
+-			label = "led_power";
++			label = "ex227:blue:power";
+ 			gpio = <&tlmm 42 GPIO_ACTIVE_HIGH>;
+ 			default-state = "on";
+ 		};
+Index: linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce/arch/arm64/boot/dts/qcom/qcom-ipq807x-ex447.dts
+===================================================================
+--- linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce.orig/arch/arm64/boot/dts/qcom/qcom-ipq807x-ex447.dts
++++ linux-4.4.60-qsdk-10fd7d14853b7020b804acae690c8acec5d954ce/arch/arm64/boot/dts/qcom/qcom-ipq807x-ex447.dts
+@@ -600,7 +600,7 @@
+ 		pinctrl-names = "default";
+ 
+ 		led_power {
+-			label = "led_power";
++			label = "ex447:blue:power";
+ 			gpio = <&tlmm 42 GPIO_ACTIVE_HIGH>;
+ 			default-state = "on";
+ 		};
-- 
2.25.1

